gym/envs/mujoco/mujoco_env.py

```diff
@@ -76,65 +84,53 @@ def viewer_setup(self):
     # -----------------------------
 
     def _reset(self):
-        mjlib.mj_resetData(self.model.ptr, self.data.ptr)
+        self.sim.reset()
         ob = self.reset_model()
         if self.viewer is not None:
-            self.viewer.autoscale()
+            # self.viewer.autoscale()
             self.viewer_setup()
         return ob
 
     def set_state(self, qpos, qvel):
         assert qpos.shape == (self.model.nq,) and qvel.shape == (self.model.nv,)
-        self.model.data.qpos = qpos
-        self.model.data.qvel = qvel
-        self.model._compute_subtree()  # pylint: disable=W0212
-        self.model.forward()
+        sim_state = self.sim.get_state()
+        sim_state.qpos[:] = qpos
+        sim_state.qvel[:] = qvel
```

Comment:
I'm pretty sure what we want to do here is
```
self.data.qpos[:] = qpos
self.data.qvel[:] = qvel
```

The state is a separate thing, and if you set those parameters on it they'll just get thrown away.